Error creating bean with name 'difyDatasetClient' defined in class path resource

### Dify version

1.1.3

### dify-spring-boot-starter version

0.8.2

### dify-spring-boot-starter used

dify-spring-boot-starter

### Steps to reproduce

启动报错，信息如下：

2025-04-17 10:11:45,894 ERROR [main] o.s.boot.SpringApplication [SpringApplication.java : 857] Application run failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'difyDatasetClient' defined in class path resource [io/github/guoshiqiufeng/dify/autoconfigure/DifyDatasetAutoConfiguration.class]: Failed to instantiate [io.github.guoshiqiufeng.dify.dataset.client.DifyDatasetClient]: Factory method 'difyDatasetClient' threw exception with message: Cannot invoke "io.github.guoshiqiufeng.dify.core.config.DifyProperties$Dataset.getApiKey()" because the return value of "io.github.guoshiqiufeng.dify.core.config.DifyProperties.getDataset()" is null
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:657)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:645)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1361)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1191)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:563)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:523)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:339)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:346)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:337)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.instantiateSingleton(DefaultListableBeanFactory.java:1155)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingleton(DefaultListableBeanFactory.java:1121)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:1056)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:987)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:318)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1361)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1350)
	at com.kind.DcicChinaTenantAppApplication.main(DcicChinaTenantAppApplication.java:27)
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.github.guoshiqiufeng.dify.dataset.client.DifyDatasetClient]: Factory method 'difyDatasetClient' threw exception with message: Cannot invoke "io.github.guoshiqiufeng.dify.core.config.DifyProperties$Dataset.getApiKey()" because the return value of "io.github.guoshiqiufeng.dify.core.config.DifyProperties.getDataset()" is null
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.lambda$instantiate$0(SimpleInstantiationStrategy.java:199)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiateWithFactoryMethod(SimpleInstantiationStrategy.java:88)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:168)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:653)
	... 21 common frames omitted
Caused by: java.lang.NullPointerException: Cannot invoke "io.github.guoshiqiufeng.dify.core.config.DifyProperties$Dataset.getApiKey()" because the return value of "io.github.guoshiqiufeng.dify.core.config.DifyProperties.getDataset()" is null
	at io.github.guoshiqiufeng.dify.autoconfigure.DifyDatasetAutoConfiguration.difyDatasetClient(DifyDatasetAutoConfiguration.java:47)
	at io.github.guoshiqiufeng.dify.autoconfigure.DifyDatasetAutoConfiguration$$SpringCGLIB$$0.CGLIB$difyDatasetClient$0(<generated>)
	at io.github.guoshiqiufeng.dify.autoconfigure.DifyDatasetAutoConfiguration$$SpringCGLIB$$FastClass$$1.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:258)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:372)
	at io.github.guoshiqiufeng.dify.autoconfigure.DifyDatasetAutoConfiguration$$SpringCGLIB$$0.difyDatasetClient(<generated>)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.lambda$instantiate$0(SimpleInstantiationStrategy.java:171)
	... 24 common frames omitted

### ✔️ Expected Behavior

_No response_

### ❌ Actual Behavior

_No response_

--- a/README-zh.md
+++ b/README-zh.md
@@ -41,7 +41,7 @@
         <dependency>
             <groupId>io.github.guoshiqiufeng.dify</groupId>
             <artifactId>dify-bom</artifactId>
-            <version>0.8.3</version>
+            <version>0.8.4</version>
             <type>import</type>
         </dependency>
     </dependencies>

--- a/README.md
+++ b/README.md
@@ -41,7 +41,7 @@ https://guoshiqiufeng.github.io/dify-spring-boot-starter/en
         <dependency>
             <groupId>io.github.guoshiqiufeng.dify</groupId>
             <artifactId>dify-bom</artifactId>
-            <version>0.8.3</version>
+            <version>0.8.4</version>
             <type>import</type>
         </dependency>
     </dependencies>

--- a/dify/dify-core/src/main/java/io/github/guoshiqiufeng/dify/core/config/DifyProperties.java
+++ b/dify/dify-core/src/main/java/io/github/guoshiqiufeng/dify/core/config/DifyProperties.java
@@ -40,12 +40,12 @@ public class DifyProperties implements Serializable {
     /**
      * 知识库
      */
-    private Dataset dataset;
+    private Dataset dataset = new Dataset();
 
     /**
      * 后台服务
      */
-    private Server server;
+    private Server server = new Server();
 
     /**
      * 请求配置

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,4 +1,4 @@
 # {x-release-please-start-version}
-APP_VERSION=0.8.3
+APP_VERSION=0.8.4
 # {x-release-please-end-version}
 APP_GROUP=io.github.guoshiqiufeng.dify

